spec/controllers/external_users/registrations_controller_spec.rb

```diff
@@ -6,122 +6,115 @@
   end
 
   describe 'POST #create' do
-    let(:email) { Faker::Internet.email }
+    subject(:perform_post) { post :create, params: { user: sign_up_attributes } }
+
     let(:sign_up_attributes) do
-      {
-        first_name: 'Bob',
+      { first_name: 'Bob',
         last_name: 'Smith',
-        email: email,
+        email: 'foo@bar.com',
         password: 'password1234',
-        password_confirmation: 'password1234'
-      }
+        password_confirmation: 'password1234',
+        terms_and_conditions: '1' }
     end
 
-    context 'when env not api-sandbox' do
-      before { post :create, params: { user: sign_up_attributes, terms_and_conditions_acceptance: '1' } }
-
-      it 'redirects to user sign up path' do
-        expect(response).to redirect_to(external_users_root_url)
-      end
-
-      it 'does not create a user' do
-        expect(User.count).to eq(0)
-      end
-
-      it 'does not create an external user' do
-        expect(ExternalUser.count).to eq(0)
+    context 'when env not api-sandbox but attributes valid' do
+      around do |example|
+        with_env('production') do
+          example.run
+        end
       end
 
-      it 'does not create a provider' do
-        expect(Provider.count).to eq(0)
-      end
+      it { expect(perform_post).to redirect_to(external_users_root_url) }
```

Comment:
nit

This could be

```ruby
it { is_expected.to redirect_to(exernal_users_root_url) }
```

and similarly in other places (although I'm not sure how it works with `expect { perform_post }`).

Reply:
`is_expected` is the same as `expect(subject)` (note the braces). `expect { perfom_post }` is a block and so cannot replaced with `is_expected`

That said, we could therefore use `is_expected` for `expect(perform_post)` but when all the the other tests use the named subject themselves I find `expect(perform_post)` clearer and more consistent - what do the other devs think? @naseberry @kmahern

Reply:
OK. Probably good to maintain consistency with the tests around it.